src/sentry/api/endpoints/project_details.py

```diff
@@ -223,6 +227,10 @@ def put(self, request, project):
             project.name = result['name']
             changed = True
 
+        if result.get('platform'):
```

Comment:
Agreed. We actually need a much more thorough VALID_PLATFORMS, something more akin to @bretthoerner's SDK matrix in code form. I don't think we should block on this, but we should make this happen. :)